parent/pom.xml

```diff
@@ -2697,6 +2697,18 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
```

Comment:
Unless there is a good reason I would avoid adding other dependencies in parent/pom

Reply:
for `xmlunit-core`, it was already used in several projects.

for `xmlunit-assertj3`, it allows to write simpler code and have a better error message

Reply:
I can update to use the same xmlunit-core in this PR, and provide a different PR to migrate from xmlunit-core to xmlunit-assertj3 to separate concerns and discuss if it is worthy or not to use the introduced one.

Reply:
Second PR provided to replace unitcore with assertj https://github.com/apache/camel/pull/11774

Reply:
the 2 other PRs have been merged.
Less code change in this PR